Nintroller/Nintroller.cs

```diff
@@ -1159,6 +1292,57 @@ private void ParseReport(byte[] report)
                 case InputReport.ExtOnly:
                     if (_state != null && _currentType != ControllerType.Unknown)
                     {
+                        // ignoring input reports arriving while guitar encryption is being set up
+                        if (_ackType.ToString().StartsWith("EncryptionSetup"))
+                        {
+                            break;
+                        }
+
+                        if (_currentType == ControllerType.Guitar)
+                        {
+                            // use the appropriate offset where extension bytes begin, based on the data report type
+                            int offset;
+                            switch (input)
+                            {
+                                case InputReport.BtnsExt:
+                                case InputReport.BtnsExtB:
+                                    offset = 3;
+                                    break;
+                                case InputReport.BtnsAccExt:
+                                    offset = 6;
+                                    break;
+                                case InputReport.BtnsIRExt:
+                                    offset = 13;
+                                    break;
+                                case InputReport.BtnsAccIRExt:
+                                    offset = 16;
+                                    break;
+                                case InputReport.ExtOnly:
+                                    offset = 1;
+                                    break;
+                                default:
+                                    return;
+                            }
+                            if (_encryptionEnabled)
+                            {
+                                GuitarDecryptBuffer(report, offset); // modify the report to contain decrypted guitar bytes
+                            }
+                            else if (BitConverter.ToString(report, offset, 6) == "00-00-00-00-00-00") // checking this only if _encryptionEnabled=false
```

Comment:
Again, would be preferred to do direct comparisons here rather than comparing against a string.